#summary use processed data to reproduce the main results shown in the manuscript

Some of the processed data are available as an Excel file (download from [https://drive.google.com/file/d/0BwieX-ApVZM5Qm05U1VaeHVYd1E/edit?usp=sharing here]) that contains Supplemental Tables S1~9 of our submitted manuscript; others can be generated using the programs provided bellow.

== analysing the nucleotide skew data ==

This section allows you use our model to :
    * analyse the nucleotide skew data, 
    * dissect contributions attributed to replication, transcription and translation, 
    * plot all main figures and figure S2,
    * do statistical tests used in the main text and supplementary text

How-to:
  # download a zip file from [https://drive.google.com/file/d/0BwieX-ApVZM5T0tSTk8zYXMyMFE/edit?usp=sharing here] and unzip it; a folder 'nucleotideSkews2014Rproject' will be created
  # launch your "R" application; if you don't have one, please download it from [http://www.r-project.org/ here]. Set the working directory to "/path/to/nucleotideSkews2014Rproject".
  # copy and paste the contents of the R source code file : 'nskews_june2014' into your R program.

*NOTE:* 
  * a R library "ggplot2" is required to make the figures; please install it if you haven't done so.
  * all required data and additional R scripts are included.

== generating the random dataset used in Figure 3 ==

The random dataset was created using a PERL script, which can be downloaded from [https://drive.google.com/file/d/0BwieX-ApVZM5ekFweWxCZ0NnT1E/edit?usp=sharing here].

*NOTE:* a PERL library "List::Util" is required.

== generating the dataset used in Figure 4 (the trade-off) ==

This dataset was created using an in-house JAVA program, which can be downloaded from [https://drive.google.com/file/d/0BwieX-ApVZM5ZFp3b083R0h0QVU/edit?usp=sharing here]. 

*NOTE:*
  # It is a NetBeans 8 project. Unzip the archived file and open it with NetBeans, you can see the source codes.
  # a precompile jar file can be found in the 'dist/' subfolder. to run it, JAVA 1.7 is required.
  # this program uses a parallel library and it will be VERY memory-consuming, so please try not to run it on your laptop / desktop.

== estimating energy investments in E. coli metabolism ==

How-to:

  # download a zip file from [https://drive.google.com/file/d/0BwieX-ApVZM5eXdyTjRSaGVTN0U/edit?usp=sharing here]
  # unzip it, a folder 'E.coli-models' will be created
  # launch your "R" application. Set the working directory to "/path/to/E.coli-models".
  # copy and paste the contents of the R source code files : 'NtAtp01.R' and "NtAtp02.R" into your R program.

* NOTE *
  # R library 'sybil' is required; see [http://www.biomedcentral.com/1752-0509/7/125 here] for the publication.
  # additional R libraries required by 'sybil' : Matrix, lattice, glpkAPI

The contents of the zip file:
  * the genome-wide metabolic model 'iJO1366' for E. coli by Orth _et al_ [http://www.ncbi.nlm.nih.gov/pubmed/21988831 (PMID: 21988831)]. 
  * two R script files.

== identifying preferred codons for selected genomes ==

[https://drive.google.com/file/d/0BwieX-ApVZM5MzBfdFZ0djFleXc/edit?usp=sharing Here] is a Java implementation of the method we used in our study.

  # It is a NetBeans 8 project. Unzip the archived file and open it with NetBeans, you can see the source codes.
  # a precompile jar file can be found in the 'dist/' subfolder. to run it, JAVA 1.7 is required.
  # this program uses a parallel library and it will be VERY memory-consuming, so please try not to run it on your laptop / desktop.
  # additional files and external programs are needed to run this program (to be further explained).